### PR TITLE
Added helpers to check for content API

### PR DESCRIPTION
**Pull request by rshbhgrg**
no issue

### CODE REVIEW

File: core/server/api/v2/utils/index.js [hunk 1]
```diff
@@ -9,5 +9,10 @@ module.exports = {
 
     get validators() {
         return require('./validators');
+    },
+
+    isContentAPI: (frame) => {
```
Comment (gargol) [inline core/server/api/v2/utils/index.js]: Looks good, would improve the readability a bit... don't have a recepie for it but maybe:
```
    isContentAPI: (frame) => {
        return !!(
            Object.keys(frame.options.context).length === 0 ||
            (!frame.options.context.user && frame.options.context.api_key_id) ||
            frame.options.context.public
        );
    }
```
what do you think @rishabhgrg ?

Comment (rishabhgrg) [reply core/server/api/v2/utils/index.js]: Looks good, I just updated with something similar, but `||` on next line. This is better, lemme update :)

File: core/server/api/v2/utils/index.js [hunk 2]
```diff
@@ -9,5 +9,12 @@ module.exports = {
...
```
Comment (kirrg001) [inline core/server/api/v2/utils/index.js]: Issue reference as comment?
https://github.com/TryGhost/Ghost/issues/10099

Comment (rishabhgrg) [reply core/server/api/v2/utils/index.js]: Updated

File: core/server/api/v2/utils/index.js [hunk 3]
```diff
@@ -9,5 +9,16 @@ module.exports = {
 
     get validators() {
         return require('./validators');
+    },
+
+    /**
+     *  TODO: We need to check for public context as permission stage overrides
+     * the whole context object currently: https://github.com/TryGhost/Ghost/issues/10099
+     */
+    isContentAPI: (frame) => {
```
Comment (kirrg001) [inline core/server/api/v2/utils/index.js]: We know it's not optimal putting this helper fn here. It should not live in index.js, but let's accept for now 👍

**Review by kirrg001 — APPROVED** [review]

File: core/test/unit/api/v2/index_spec.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        it('is truthy when context is empty', function () {
+            const frame = {
+                options: {
+                    context: {
+                    }
+                }
+            };
+            should(utils.isContentAPI(frame)).equal(true);
+        });
+    });
```
Comment (gargol) [inline core/test/unit/api/v2/index_spec.js]: @rishabhgrg one more case missing for:
```
            const frame = {
                options: {
                    context: {
                        public: true
                    }
                }
            };
```
:wink:

**Comment by rishabhgrg** [comment]
@kirrg001 / @gargol : Should we also add the helper `isContentAPI` to test utils separately here? Or we use it directly from here in tests?

File: core/test/unit/api/v2/index_spec.js (hunk 4 above)
Comment (rishabhgrg) [reply core/test/unit/api/v2/index_spec.js]: Updated 👍

**Comment by gargol** [comment]
@rishabhgrg, could you expand on why do we need it as a helper in tests?

**Comment by rishabhgrg** [comment]
@gargol 🤔 No I thinks its my bad, my initial thought was it will be needed for input serializer tests, like [here](https://github.com/TryGhost/Ghost/pull/10098/files#diff-9d14902f9b28ef5a9a579373ab23c185R52) to check if its content api, but that can be done by configuring the `frame` options  I think now so we don't need it. 😅

**Pull request merged**